chrome/content/zotero/zoteroPane.js

```diff
@@ -3505,6 +3509,23 @@ var ZoteroPane = new function()
 			() => collection.addItems(items.map(item => item.id)));
 	};
 
+
+	this.addTagToSelectedItems = function () {
+		let items = Zotero.Items.keepParents(this.getSelectedItems());
+		let out = { value: null };
+		let result = Services.prompt.prompt(
```

Comment:
People are going to want to use this in lieu of dragging, particularly if there's a keyboard shortcut, so I think this should be a nice big popup panel that does tag autocomplete:

https://github.com/zotero/zotero/blob/0785b3a0864fc2dfce6ab53a88c08f6d9d1cd5a7/chrome/content/zotero/containers/tagsBoxContainer.jsx#L71-L99

For the input, we have this — with a `getSuggestions` prop — in React, which is what we use in the tagsBox component:

https://github.com/zotero/zotero/blob/master/chrome/content/zotero/components/form/input.jsx

The prompt works, of course, but this hasn't existed for 15 years, so I think we should do it right — major new version, major new way of assigning quickly tags to one or more items with a keyboard shortcut.

We might also consider doing this for collections, with results showing the "path" to subcollections, instead of the hierarchical menu, which gets pretty unwieldy in a huge library. Or at least offering it as a keyboard-based alternative.

(There's no rush on this.)